packages/bitcore-node/src/modules/ripple/api/wallet-tx-transform.ts

```diff
@@ -33,78 +33,76 @@ export class RippleDbWalletTransactions extends Transform {
       .addCursorFlag('noCursorTimeout', true)
       .toArray();
 
-    const walletAddresses = await this.getAddresses();
-    const relevantAddresses = walletAddresses.map(w => w.address);
-    let sending = relevantAddresses.includes(tx.from);
-    for (const output of outputs) {
-      const { address } = output;
-      let receiving = false;
-      if (relevantAddresses.includes(output.address) && output.value > 0) {
-        receiving = true;
-      }
-      if (sending) {
-        if (!receiving) {
-          this.push(
-            JSON.stringify({
-              id: tx.txid,
-              txid: tx.txid,
-              fee: tx.fee,
-              size: 0,
-              category: 'send',
-              satoshis: -1 * Math.abs(output.value) + tx.fee,
-              height: tx.blockHeight,
-              address,
-              outputIndex: output.mintIndex,
-              blockTime: tx.blockTimeNormalized
-            }) + '\n'
-          );
-        } else {
-          this.push(
-            JSON.stringify({
-              id: tx.txid,
-              txid: tx.txid,
-              fee: tx.fee,
-              size: 0,
-              category: 'move',
-              satoshis: -1 * Math.abs(output.value),
-              height: tx.blockHeight,
-              address,
-              outputIndex: output.mintIndex,
-              blockTime: tx.blockTimeNormalized
-            }) + '\n'
-          );
-        }
-        if (tx.fee > 0) {
-          this.push(
-            JSON.stringify({
-              id: tx.txid,
-              txid: tx.txid,
-              category: 'fee',
-              satoshis: -1 * tx.fee,
-              height: tx.blockHeight,
-              blockTime: tx.blockTimeNormalized
-            }) + '\n'
-          );
-        }
-        return done();
-      } else {
-        if (receiving) {
-          this.push(
-            JSON.stringify({
-              id: tx.txid,
-              txid: tx.txid,
-              fee: tx.fee,
-              size: 0,
-              category: 'receive',
-              satoshis: Math.abs(output.value),
-              height: tx.blockHeight,
-              address,
-              outputIndex: output.mintIndex,
-              blockTime: tx.blockTimeNormalized
-            }) + '\n'
-          );
-        }
+    const walletAddressesObjs = await this.getAddresses();
+    const walletAddresses = walletAddressesObjs.map(w => w.address);
+    let sending = walletAddresses.includes(tx.from);
+    if (sending) {
+      const sends = outputs.filter(o => !walletAddresses.includes(o.address));
```

Comment:
What you think about defining sends/moves/receives on a single loop on outputs, so we can optimize the flow a bit?

Reply:
For this diff, it was originally a single loop, but the logic was confusing to think through.

I can rewrite it as a single loop again, and get rid of the filters and forEach